### PR TITLE
Ontology reverse id label yaml config

### LINKED ISSUE #313: Ontology: Make reverse_labels configurable via biocypher_config.yaml
Do I understand the code correctly, that at the moment it is not possible to define in the `biocypher_config.yaml`, whether the node id and label should be switched (it is default to `true`).

It would be convenient to have the possibility to optionally define it in the `biocypher_config.yaml` and then pass on the defined value. 
```
  head_ontology:
    url: config/ontologies/custom.ttl
    root_node: Ontology
    switch_label_and_id: <true, false>
```

If this is desired functionality, I will create a PR for this.

### LINKED ISSUE #319: Multiple copies of NetworkX graph in memory: problem for larger ontologies?
https://github.com/biocypher/biocypher/blob/06389f5c96675cb19700f4abb9aee5af8e18ca52/biocypher/_ontology.py#L110

In this method, we create multiple iterations of the initial graph object; does this create an overhead? Do we need to consider removing some of the duplications?

### LINKED ISSUE #320: Head join node not found in head ontology
When loading two very simple Turtle ontologies made from scratch in Protégé, configuring a tail-in fails on: "ValueError: Head join node <whatever> not found in head ontology."

## Details

```
INFO -- This is BioCypher v0.5.37.
INFO -- Logging into `biocypher-log/biocypher-20240304-214431.log`.
INFO -- Loading ontologies...
INFO -- Instantiating OntologyAdapter class for ontology1.ttl.
INFO -- Instantiating OntologyAdapter class for ontology2.ttl.
Traceback (most recent call last):
  File "/home/cl/travail/code/onto/tests/main.py", line 8, in <module>
    bc.show_ontology_structure()
  File "/home/cl/travail/code/biocypher/biocypher/_core.py", line 557, in show_ontology_structure
    self._get_ontology()
  File "/home/cl/travail/code/biocypher/biocypher/_core.py", line 199, in _get_ontology
    self._ontology = Ontology(
                     ^^^^^^^^^
  File "/home/cl/travail/code/biocypher/biocypher/_ontology.py", line 402, in __init__
    self._main()
  File "/home/cl/travail/code/biocypher/biocypher/_ontology.py", line 415, in _main
    self._assert_join_node(adapter)
  File "/home/cl/travail/code/biocypher/biocypher/_ontology.py", line 467, in _assert_join_node
    raise ValueError(
ValueError: Head join node entity not found in head ontology.
```

main.py:
```python
import biocypher

bc = biocypher.BioCypher(
    biocypher_config_path = "biocypher.yaml",
    schema_config_path = "schema.yaml"
)
bc.show_ontology_structure()
```

biocypher.yaml:
```yaml
biocypher:
    debug: true
    offline: true

    head_ontology:
        url: ontology1.ttl
        root_node: Thing

    tail_ontologies:
        urref:
            url: ontology2.ttl
            head_join_node: entity
            tail_join_node: EvaluationCriterion
```

schema.yaml:
```yaml
entity:
    represented_as: node
    label_in_input: entity

Accuracy:
    represented_as: node
    label_in_input: Accuracy

user_is_accurate:
    is_a: relation
    represented_as: edge
    label_in_input: user_is_accurate
    source: User
    target: EvaluationCriterion
```

Ontologies:
[test_2ontologies.zip](https://github.com/biocypher/biocypher/files/14487006/test_2ontologies.zip)

## Configuration

- Linux Ubuntu 23.10 (mantic)
- Protégé v5.6.3
- Installed via poetry 1.6.1
- BioCypher v0.5.37
- Python 3.11.6
- poetry.lock: [poetry.lock.txt](https://github.com/biocypher/biocypher/files/14487069/poetry.lock.txt)

### PR DESCRIPTION
**Pull request by nilskre**
Closes #313 
Closes #319
Closes #320 
- #313 -> switch_label_and_id can now also be configured via the yaml config
- #319 -> remove the assignment to new objects (now only one netwokx graph object)
- #320 -> improve error messages (log defined node and possible nodes in the ontology)

### CODE REVIEW

File: biocypher/_ontology.py [hunk 1]
```diff
@@ -514,15 +566,37 @@ def _assert_join_node(self, adapter: OntologyAdapter) -> None:
                 join node in the head ontology.
         """
 
-        head_join_node = adapter.get_head_join_node()
+        head_join_node = None
+        head_join_node_label = adapter.get_head_join_node().replace("_", " ")
```
Comment (slobentanzer) [inline biocypher/_ontology.py:572]: In other methods, we make replacing underscores contingent on the `rename_nodes` parameter; why do we not do that here? We also convert to lowercase in the other places when we rename, why not here?

**Review by slobentanzer — COMMENTED** [review]
Great work! Just one minor question.

**Comment by nilskre** [comment]
Good spotted point. I have acted upon it and it should be ready now. @slobentanzer

**Review by slobentanzer — APPROVED** [review]
Great, thanks!

**Pull request merged**